### PR TITLE
Send alert for requests needing debt data after ledger split

### PR DESCRIPTION
**Pull request by marctemp**
https://eaflood.atlassian.net/browse/SFI-1997

### CODE REVIEW

**Comment by marctemp** [comment]
Split 1 function needed into multiple files as a pre-successor to the tidy up given in these tech debt tickets:

- https://eaflood.atlassian.net/browse/SFI-1953
- https://eaflood.atlassian.net/browse/SFI-1998

File: app/debt/check-debts-by-enrichment.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const checkDebtsByEnrichment = async (frn, reference, netValue, transaction) =>
   const parsedFrn = parseInt(frn)
 
   if (isNaN(parsedFrn)) {
-    return {}
+    return null
```
Comment (marctemp) [inline app/debt/check-debts-by-enrichment.js:7]: https://sequelize.org/docs/v6/core-concepts/model-querying-finders/#findone

File: app/manual-ledger/check-for-ar-ledger.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  if (arLedger) {
+    const debtData = await checkForDebtData(arLedger)
+    if (!debtData) {
+      console.log('no debt data found')
+      status = AWAITING_ENRICHMENT
+      if (config.isAlerting) {
+        await sendEnrichRequestBlockedEvent(manualLedgerRequest[0])
+      }
+    }
```
Comment (marctemp) [inline app/manual-ledger/check-for-ar-ledger.js:23]: Only actual functionality added, rest is migrating to different modules for testing

**Comment by marctemp** [comment]
Screenshot 2022-06-06 at 11 53 58


Also, getting an unusual difference with the GOV UK frontend not rendering.

Looked into the node module not being loaded but would cause issues elsewhere, functionality works fine

File: app/manual-ledger/check-for-ar-ledger.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  const arLedger = manualLedgerRequest
+    .find(x => x.ledgerPaymentRequest?.ledger === AR && x.ledgerPaymentRequest?.value !== 0)
+
+  if (arLedger) {
+    const debtData = await checkForDebtData(arLedger)
+    if (!debtData) {
+      console.log('no debt data found')
+      status = AWAITING_ENRICHMENT
+      if (config.isAlerting) {
+        await sendEnrichRequestBlockedEvent(manualLedgerRequest[0])
```
Comment (AbidemiAdio) [inline app/manual-ledger/check-for-ar-ledger.js:21]: Consider arLedger as the parameter for sendEnrichRequestBlockedEvent but not manualLedgerRequest[0] : - 
 await sendEnrichRequestBlockedEvent(arLedger)

**Review by AbidemiAdio — DISMISSED** [review]
QED

**Review by johnwatson484 — APPROVED** [review]

**Pull request merged**